packages/hawtio/src/core/event-service.ts

```diff
@@ -49,30 +58,38 @@ class EventService implements IEventService {
     log.debug('Number of listeners on', EVENT_NOTIFY, '=', this.eventEmitter.listenerCount(EVENT_NOTIFY))
   }
 
-  login(): void {
+  login() {
     this.eventEmitter.emit(EVENT_LOGIN)
   }
 
-  onLogin(listener: EventListener): void {
+  onLogin(listener: EventListener) {
     this.eventEmitter.on(EVENT_LOGIN, listener)
   }
 
-  logout(): void {
+  logout() {
     this.eventEmitter.emit(EVENT_LOGOUT)
   }
 
-  onLogout(listener: EventListener): void {
+  onLogout(listener: EventListener) {
     this.eventEmitter.on(EVENT_LOGOUT, listener)
   }
 
-  refresh(): void {
+  refresh() {
     this.eventEmitter.emit(EVENT_REFRESH)
   }
 
-  onRefresh(listener: EventListener): void {
+  onRefresh(listener: EventListener) {
     this.eventEmitter.on(EVENT_REFRESH, listener)
   }
 
+  pluginsUpdated() {
```

Comment:
With the introduction of this event listener and response, it maybe useful to now update the `workspace` response to `pluginsUpdated` [here](https://github.com/hawtio/hawtio-next/blob/main/packages/hawtio/src/plugins/shared/workspace.ts#L66) as a `window.reload()` is going to make all other responses to `pluginsUpdated` pointless, I'd have thought?

Reply:
Good catch. Yes, that's also what I had in mind a bit as well. We'd need to review consistency between the new plugin system and the watcher impl we have in the workspace.

In reality, probably that plugin watcher MBean is an old mechanism, where in the past we expected dynamic additions and deletions of plugins via JMX, especially on dynamic environments like OSGi/Karaf. But now, with Hawtio.next we currently only support Spring Boot and Quarkus as the runtimes and they are relatively static at runtime; all dependencies would be resolved before bootstrap. I can't think of use cases where plugins are added dynamically after application launches on Spring Boot and Quarkus. Correct me if I'm wrong.

This plugin implementation here only loads external plugins at the initial page loading time, and it might be enough for the runtimes Hawtio.next supports. If that's the case, we can consider just removing the plugin watcher from hawtio-next and the main hawtio project.

Reply:
Filed follow up issue #214